docs/xrpl-wallet/api-reference/topup.mdx

```diff
@@ -0,0 +1,89 @@
+---
+title: Topup
+---
+
+Top-up your cryptocurrency through our fiat-to-crypto partners with ease.
+
+---
+
+## `initiateTopup`
+
+Use the `initiateTopup` function to start a top-up transaction with the selected provider from the dapp. Check the list below for supported values of parameters and providers.
+
+:::info NOTE
+
+You don't need to be logged in with Torus to use the top-up API.
+
+:::
+
+```javascript
+const paymentStatus = await torus.initiateTopup(provider, params);
+```
+
+**Parameters**
+
+| Name     | Type                    | Description                                                               | Mandatory |
+| :------- | :---------------------- | :------------------------------------------------------------------------ | :-------- |
+| provider | `PAYMENT_PROVIDER_TYPE` | The specified payment provider partner.                                   | Yes       |
+| params   | `PaymentParams`         | The topup tx params. used to autofill the form for that specific provider | Yes       |
+
+**Returns**
+
+- `Promise<boolean>` : Returns a promise which resolves to a `boolean` indicating whether the user has successfully completed the top-up flow.
+
+### `PAYMENT_PROVIDER_TYPE`
+
+The list of `PAYMENT_PROVIDER_TYPE`:
+
+- `topper`
+
+### `PaymentParams`
+
+**Parameters**
+
+| Name                   | Type     | Description                                               | Mandatory |
+| :--------------------- | :------- | :-------------------------------------------------------- | :-------- |
+| selectedAddress        | `string` | Address to send the funds to                              | No        |
+| selectedCurrency       | `string` | Default fiat currency for the user to make the payment in | No        |
+| selectedCryptoCurrency | `string` | Cryptocurrency to buy                                     | No        |
```

Comment:
selectedCryptoCurrency should be XRP always for now, we don't support any other in this wallet.